applicationset/generators/matrix.go

```diff
@@ -38,44 +40,49 @@ func (m *MatrixGenerator) GenerateParams(appSetGenerator *argoprojiov1alpha1.App
 		return nil, EmptyAppSetGeneratorError
 	}
 
-	if len(appSetGenerator.Matrix.Generators) < 2 {
+	numGens := len(appSetGenerator.Matrix.Generators)
+	if numGens < 2 {
 		return nil, ErrLessThanTwoGenerators
 	}
 
-	if len(appSetGenerator.Matrix.Generators) > 2 {
-		return nil, ErrMoreThanTwoGenerators
+	if m.maxChildren > 1 && numGens > m.maxChildren {
```

Comment:
When maxChildren is 1 or less, it's considered no limit. The default value for maxChildren is 2.

Reply:
Not a showstopper but That sounds un-intuitive IMO - could we not say that no limit only applies if maxChildren < 0?

Reply:
I would prefer to have maxChildren < 1 in that case to avoid using negative values. Other params, such as `reposerver.git.lsremote.parallelism.limit` or `controller.k8sclient.retry.max`, have the same logic where any values less than 1 mean the max or unlimited.